Name initial core team and add documentation about roles

Zsailer opened the pull request:
Follow up to #5.

Names the core team based on responses from #5. Adds a set of MyST pages and config for rendering via readthedocs. 

The pages define the responsibilities and privileges of a team member. This is almost entirely borrowed from the [JupyterHub/Binder Team Compass](https://github.com/jupyterhub/team-compass) repo.

blink1073 commented on docs/team/governance.rst:
```diff
@@ -0,0 +1,131 @@
+   promoting the Jupyter Server Project. You are expected to coordinate your actions with
+   other red members so that the project presents a coherent position
+   towards the outside. You are expected to prepare and attend the
+   monthly team calls.
```
weekly?

Zsailer replied:
```suggestion
   weekly team calls.
```

blink1073 commented on docs/team/governance.rst:
```diff
@@ -0,0 +1,131 @@
+-  **Green member.** Your life situation has changed so that you prefer to
+   (temporarily) not take on the rights and responsibilities of being an
+   active member. You can return to active membership by resuming your
+   community activities. Your return will be announced at the next monthly
```
weekly?

Zsailer replied:
```suggestion
   community activities. Your return will be announced at the next weekly
```

blink1073 reviewed (approved):
Thank you!

mwakaba2 commented on docs/team/contributors-jupyter-server.yaml:
```diff
@@ -0,0 +1,62 @@
+  team: red
+  last-check-in: 2021-12
+
+- name: Mariki Wakabayashi
```
Could you update it to `Mariko`? 🙂

Zsailer replied:
🤦 Oh my goodness. So sorry for the typo, Mariko! 😅

Zsailer replied:
```suggestion
- name: Mariko Wakabayashi
```

kevin-bates replied:
Looks like this got revived on the redo.

vidartf commented:
Was the .DS_store meant to be committed?

vidartf commented on docs/team/governance.rst:
```diff
@@ -0,0 +1,131 @@
+   promoting the Jupyter Server Project. You are expected to coordinate your actions with
+   other red members so that the project presents a coherent position
+   towards the outside. You are expected to prepare and attend the
+   weekly team calls.
```
As mentioned in the weekly meeting today, we might want to relax this sentence a bit so that we also can accept members whose schedule do not allow them to participate in meetings (different timezone, other responsibilities, work-life balance), but who are still able to engage in the running of the project asynchronously.

Zsailer replied:
Updated

kevin-bates commented on docs/team/contributors-jupyter-server.yaml:
```diff
@@ -0,0 +1,62 @@
+- name: Kevin Bates
+  handle: "@kevin-bates"
+  affiliation: IBM
+  team: red
```
Per my recent comment...
```suggestion
  team: green
```

Zsailer replied:
:+1: Thanks, Kevin. Let us know when you're ready to join as Red Team again!

Zsailer commented:
I've removed myself from "Lead" to Red Team for this PR. I think it's most appropriate to nominate a Team Lead in its own PR and merge it after the initial team is formed. We can do this early in the new year.

This PR should be ready for final review.

kevin-bates reviewed (approved):
The built output looks great @Zsailer - thank you!

lresende commented:
I know you are trying to follow the JupyterHub notion of team colors, but I would like to really ask yourself if this would introduce a barrier to growing the community. Most open-source projects today have a notion of committers/maintainers and core maintainers/PMC (or emeritus) which works without the need of a legend, which is not true for the "group colors". I also think the membership is based on trust, and we should not distinguish between red and blue teams, basically, you move a step on the latter based on your contributions, being those code or non-code... but I wouldn't argue about that. 

Another issue is that, with the new governance, we are moving away from BDFL favoring consensus building with a process to reconcile disagreements via vote, and we should really avoid naming a `team leader` which implies that that person has a different level of power among the community members.

lresende commented:
Also, if the folks are planning to move [Jupyter Enterprise Gateway](https://github.com/jupyter/governance/pull/110) to this org, I would consider adding @lresende and @kevin-bates as core-committers.

echarles commented:
> Most open-source projects today have a notion of committers/maintainers and core maintainers/PMC (or emeritus) which works without the need of a legend, which is not true for the "group colors". I also think the membership is based on trust, and we should not distinguish between red and blue teams, basically, you move a step on the latter based on your contributions, being those code or non-code... but I wouldn't argue about that.

I agree with @lresende that I need the legend to make sense of the colors, and that I would feel more comfortable with Contributor/Committer/PMC/Chair/Emeritus status as I have been used when being a active Apache committer (https://people.apache.org/committer-index.html, maybe also influencing @lresende as being also a Apache committer).

This is a discussion that can anyway occur in the future in relationship what the other Jupyter projects are doing, and based on Board inputs, so for the sake of moving forward the team setup, I want to approve this PR, but will wait a bit to ensure @lresende is added to the Team.

Thx All!

Zsailer commented:
Thank you for your feedback, @lresende and @echarles! 

I broke up my response into two separate decisions to make:

1. Committer/PMC/Emeritus model vs. Red/Blue/Green model
2. Eliminate Team Lead Role (keep SSC representative role, of course) 

### Committer/PMC/Emeritus model vs. Red/Blue/Green model

> Most open-source projects today have a notion of committers/maintainers and core maintainers/PMC (or emeritus) which works without the need of a legend, which is not true for the "group colors". 

I think you're right. I think this terminology might be more broadly familiar in open-source. I'm happy to drop the team colors in favor of the committer, PMC, emeritus language if others agree..

> this would introduce a barrier to growing the community

I'm not sure that I agree that this model introduces more of a barrier than the Apache model, except that people might be less familiar with the colors' meanings.

I think the intentions behind the Red/Blue/Green model from JupyterHub is that it encouraged a broader contributor community by recognizing non-code contributors as team members (with write access). In my experience the committer/PMC/emeritus model tends to select mostly for code contributors *only*.

> we should not distinguish between red and blue teams, basically, you move a step on the latter based on your contributions, being those code or non-code... but I wouldn't argue about that.

Hm, isn't this similar to distinguishing between committer and PMC in Apache's model? I think the Blue -> Red transition is less restrictive, because it doesn't require a nomination from other Red members; rather you just re-label yourself based on your contributions. In Apache's model, you need to be nominated for committer, then nominated again, later, for PMC.

It's possible we are just discussing semantics here, but I think it's worth discussing. 

### Eliminate Team Lead Role

> consensus building with a process to reconcile disagreements via vote, and we should really avoid naming a team leader which implies that that person has a different level of power among the community members.

This is a fantastic point, and I wholeheartedly agree. You'll see in #5 that I conflated the Software Steering Council (SSC) representative with the Team Lead role. 

I propose that we should drop the notion of "Team Lead" altogether and just have a nominated SSC representative role. They don't have any decision making power; rather, they represent the Jupyter Server subproject in an SSC vote.

Zsailer commented:
@lresende, I've added you to the team 😎.

SylvainCorlay commented on docs/team/contributors-jupyter-server.yaml:
```diff
@@ -0,0 +1,68 @@
+  handle: "@jtpio"
+  affiliation: Quantstack
+  team: green
+  last-check-in: 2021-12
```
```suggestion
  last-check-in: 2021-12
  
  - name: Sylvain Corlay
    handle: "@SylvainCorlay"
    affiliation: QuantStack
    team: green
    last-check-in: 2022-01
```

SylvainCorlay commented on docs/team/contributors-jupyter-server.yaml:
```diff
@@ -0,0 +1,68 @@
+  handle: "@lresende"
+  affiliation: Apple
+  team: red
+  last-check-in: 2021-12
```
```suggestion
  last-check-in: 2021-12
  
  - name: Sylvain Corlay
  handle: "@SylvainCorlay"
  affiliation: QuantStack
  team: red
  last-check-in: 2022-01
```

SylvainCorlay commented on docs/team/contributors-jupyter-server.yaml:
```diff
@@ -0,0 +1,68 @@
+  handle: "@lresende"
+  affiliation: Apple
+  team: red
+  last-check-in: 2021-12
```
```suggestion
  last-check-in: 2021-12
  
 - name: Sylvain Corlay
   handle: "@SylvainCorlay"
   affiliation: QuantStack
   team: red
   last-check-in: 2022-01
```

Zsailer replied:
:+1:

Zsailer commented:
We discussed this PR quite a bit in the Jupyter Server meeting today. I will be making several changes based on that conversation:
* Absolve the team lead position
* Eliminate the team colors.
* Define two categories of team members, "active" and "not currently active". Active team members have voting rights and not active members do not.
* Add description of SSC Representative to docs.

Zsailer commented:
As a result, everyone who asked to be a red+blue team member will be added to the active list. Green members will be added to the "not currently active" list.

Zsailer commented:
Okay, I've rewritten the docs quite extensively based on the feedback from today. Please feel free to review and leave feedback.

I've also changed the team to use Sphinx Book Theme, since that seems better for this type of documentation.

We can leave this PR open for one more week and merge it at next week's meeting. Thank you all for a super helpful discussion.

kevin-bates commented on docs/team/member-guide.md:
```diff
@@ -0,0 +1,111 @@
+   When creating issues, provide enough context so that others can understand
+   and provide their input. Encourage feedback and input from others
+   often, and be patient when merging code - it is almost better to
+   wait a bit for an approve than to self-merge.
```
typo
```suggestion
   wait a bit for an approval than to self-merge.
```

kevin-bates commented on docs/team/member-guide.md:
```diff
@@ -0,0 +1,111 @@
+### General policy about communication channels
+
+We are trying to organize our discussions in order to help both contributors and
+maintainers find and choose the right communication channels and have a positive experience :-)
```
The `:-)` seems a little out of place.  Might be best to just replace it with a period.  I believe the tone communicates the intention of the emoticon already.

Zsailer replied:
:+1:

kevin-bates reviewed (approved):
This looks good Zach - thank you!  Just had a couple of nits and a mention regarding Mariko's name.

Zsailer commented on docs/team/member-guide.md:
```diff
@@ -0,0 +1,111 @@
+### General policy about communication channels
+
+We are trying to organize our discussions in order to help both contributors and
+maintainers find and choose the right communication channels and have a positive experience :-)
```
```suggestion
maintainers find and choose the right communication channels and have a positive experience.
```

Zsailer commented on docs/team/becoming-member.md:
```diff
@@ -0,0 +1,68 @@
+
+Team members can freely pass between active and inactive at any time. They *should* publicly state their status change in an issue or PR. A subsequent pull request will be made to this page to update their status.
+
+This means an inactive team member can "reactivate" themselves at any time by publically stating their change in status. This does not require a nomination from another team member.
```
```suggestion
This means an inactive team member can "reactivate" themselves at any time by publicly stating their change in status. This does not require a nomination from another team member.
```

jweill-aws commented on docs/team/member-guide.md:
```diff
@@ -0,0 +1,111 @@
+4. **Let us know if you'll be unavailable** or out of town for an extended period
+   of time. It's no problem if you need to focus on other things for a bit, but it's
+   helpful for the team to know who will be around.
+   To do so, [open an issue in the team-compass](https://github.com/jupyter-server/team-compass/issues/new).
```
A shared calendar (e.g., on Google Calendar) might make it easier for people to see, at a glance, when team members are unavailable. Grant all team members permissions to add their own events to the calendar. At a previous company, our team used one for this purpose.

kevin-bates replied:
Is this implying that the team member will be transitioning to 'inactive'?  Or just indicating they'll be away for a bit but intend to remain 'active'.  Opening an issue to indicate they won't be at the next meeting or two (or very active in the org for a couple of weeks) seems a little heavyweight.  I like @jweill-aws's suggestion of an internal calendar - available only to team members - for this purpose.

echarles replied:
I have read a few durations (3 weeks, 1 year) regarding the expected period to declare being inactive. On my side, I'd prefer not having to ping-pong between active and inactive based on my schedule. I'd also prefer not to add yet another calendar to maintain. My previous experience at Apache is that once you get committer role, you get it for life, whatever happens. A committer can retire him self if he wants. The exception is the Apache membership where you are expected to participate to the yearly vote.

For here, I would prefer that we say to bump to inactive for year order of magnitude inactivity.

Zsailer replied:
Let's drop the part where we say to open an issue. I agree, this is heavy handed. This can happen informally. 

Really the goal here is to relieve a team member from feeling pressured to watch this repo for any formal votes that arise while they are out on leave/vacation etc.

mwakaba2 commented on docs/team/becoming-member.md:
```diff
@@ -0,0 +1,68 @@
+
+This means an inactive team member can "reactivate" themselves at any time by publicly stating their change in status. This does not require a nomination from another team member.
+
+For example, a team member who is going out on a long leave/vacation can temporarily move to the inactive team during their absence and immediately reactivate upon return.
```
what would be considered a long leave or vacation? It might help to state the duration or some kind of minimum threshold. (e.g. anything longer than 2 or 3 weeks)

Zsailer replied:
Good question. Really, this is just a way for a team-member to say they wouldn't be available for any votes that arise during the period they are away. 

Since voting cycles are about 2 weeks long (based on Jupyter governance), 2 weeks is probably an appropriate duration to use here.

I'm not sure we need to be this technical, but perhaps this would be a good guide.

mwakaba2 reviewed (approved):
looks great @Zsailer ! Thank you for putting this together! 
I just had one question around what's considered a long leave or vacation.

blink1073 commented on docs/team/becoming-member.md:
```diff
@@ -0,0 +1,68 @@
+* Can be elected as the Software Steering Council representative for Jupyter Server.
+* Get a vote in a voting situation.
+* Count towards quorum in a voting situation.
+* Expected to participate in a majority of the team votes.
```
```suggestion
* Are expected to participate in a majority of the team votes.
```

blink1073 commented on docs/team/becoming-member.md:
```diff
@@ -0,0 +1,68 @@
+
+Inactive members:
+
+* Previously an active team member
```
```suggestion
* Were previously an active team member
```

blink1073 commented on docs/team/becoming-member.md:
```diff
@@ -0,0 +1,68 @@
+Inactive members:
+
+* Previously an active team member
+* Does not vote.
```
```suggestion
* Do not vote.
```

blink1073 commented on docs/team/member-guide.md:
```diff
@@ -0,0 +1,111 @@
+   so that you're notified when team conversations are happening.
+2. **Stay up-to-date on team meetings**. You can find a notes from previous meetings pinned at
+   the [top of the team-compass issues page](https://github.com/jupyter-server/team-compass/issues).
+3. **Vote**. Participate in at least 2/3 of votes happening in the team-compass repo. You should be   automatically pinged on Github when a vote is called.
```
```suggestion
3. **Vote**. Participate in at least 2/3 of votes happening in the team-compass repo. You should be automatically pinged on Github when a vote is called.
```

kevin-bates commented on docs/team/becoming-member.md:
```diff
@@ -0,0 +1,68 @@
+* Are not counted towards voting quorum.
+* Can "reactivate" at any time by expressing their change in status publicly.
+
+Team members can freely pass between active and inactive at any time. They *should* publicly state their status change in an issue or PR. A subsequent pull request will be made to this page to update their status.
```
Would it be more straightforward to just open a PR against the `contributors-jupyter-server.yaml` file and skip the interim "issue or PR"?
```suggestion
Team members can freely pass between active and inactive at any time. They *should* publicly state their status change in a pull request that updates the `contributors-jupyter-server.yaml` file with their status change.
```
(With a link to the file.)

kevin-bates commented on docs/team/becoming-member.md:
```diff
@@ -0,0 +1,68 @@
+positive, productive member of the community. Newcomers are encouraged to
+become team members, after they've shown a sustained interest in
+engaging with the community. Moreover, team members should be interested in
+**continuing their engagement** over a long-ish period of time, generally
```
Should probably quantify "long-ish"...  Perhaps one year?
```suggestion
**continuing their engagement** over a long-ish period of time (at least one year), generally
```

kevin-bates commented on docs/team/becoming-member.md:
```diff
@@ -0,0 +1,68 @@
+mean contributing code - it can be assisting others in forums/issues, reviewing
+pull requests, participating in team meetings, etc.
+
+Any new team members must be nominated and championed by a pre-existing team member.
```
```suggestion
Any new team members must be nominated and championed by an active team member.
```

kevin-bates commented on docs/team/becoming-member.md:
```diff
@@ -0,0 +1,68 @@
+Any new team members must be nominated and championed by a pre-existing team member.
+This process takes the following steps:
+
+1. The champion should first discuss internally with team members to get buy-
+   in and ensure that there's general consensus before officially starting
+   the process.
```
```suggestion
1. The champion should first discuss internally with team members to 
   ensure that there's general consensus before officially starting
   the process.
```

kevin-bates commented on docs/team/becoming-member.md:
```diff
@@ -0,0 +1,68 @@
+   in and ensure that there's general consensus before officially starting
+   the process.
+2. If there seems to be team consensus,
+   the champion contacts the potential new team member and ask if they are
```
```suggestion
   the champion contacts the potential new team member and asks if they are
```

kevin-bates commented on docs/team/becoming-member.md:
```diff
@@ -0,0 +1,68 @@
+
+For someone to become a team member, they should already be a consistent,
+positive, productive member of the community. Newcomers are encouraged to
+become team members, after they've shown a sustained interest in
```
```suggestion
become team members after they've shown a sustained interest in
```

kevin-bates reviewed (commented):
Mostly nits.  I also agree with previous comments.

blink1073 commented on docs/team/becoming-member.md:
```diff
@@ -0,0 +1,68 @@
+
+Inactive members:
+
+* Were previously an active team member
```
```suggestion
* Were previously an active team member.
```

Zsailer commented:
## January 13th, 2022

### Attendees

| Name          | affiliation                 | username         |
| ------------- | --------------------------- | -----------------|
| Zach Sailer | Apple | @Zsailer | 
| Jason Weill | AWS | @jweill-aws |
| Vidar T Fauske | JP Morgan Chase | @vidartf | 
| David Brochart | QuantStack | @davidbrochart |
| Sylvain Corlay | QuantStack | @SylvainCorlay |
| A. T. Darian | Two Sigma | @afshin |
| Kevin Bates | IBM | @kevin-bates |
| Max Klein | JP Morgan Chase | @telamonian |
| Mariko Wakabayashi | OpenZeppelin | @mwakaba2 |
| Eric Charles | Datalayer | @echarles |



### Agenda + Minutes

* [Protocol Alignment Proposal](https://github.com/jupyter-server/jupyter_server/pull/657)
    * See the PR for full description. Kudos to David for a clear, well-written proposal. :tada:
    * We need to support both protocols (old and new). We can change the default in a major release sometime in the future.
    * Question from the group: do we use a versioned endpoint or query parameter to select the protocol version? 
    * It might be possible to use websocket protocols fields to select the protocol: https://developer.mozilla.org/en-US/docs/Web/API/WebSocket/WebSocket https://html.spec.whatwg.org/multipage/web-sockets.html#the-websocket-interface,  there is a question if this could be made backwards-compatible though.
* [One 0MQ channel per kernel](https://github.com/jupyter-server/jupyter_server/issues/658)
* Jupyter Server 1.13.2 released this week.
    * https://github.com/jupyter-server/jupyter_server/releases/tag/v1.13.2
* Finalizing the Jupyter Server Team
    * Outstanding comments/questions.
    * https://github.com/jupyter-server/team-compass/pull/14
* Pending Kernels update
    * Starting, Shutting down, restarting, and interrupting are all pending states.
    * PR in Jupyter Client https://github.com/jupyter/jupyter_client/pull/732
    * Updated Jupyter Server Tests: https://github.com/jupyter-server/jupyter_server/pull/654
* Questions about jupyter_server/jupyter_client
    * In the case of ZMQ over TCP, how are the sockets connected?
    * What part of the code consumes the connection file?
        * In local model, these are managed and consumed by local kernel provisioners
* KSMM (Kernelspec Manager) Demo
    * https://github.com/deshaw/ksmm
    * Demo from Eric Charles
    * Parameterize Kernels by generating static kernelspecs using a templating engine
    * Combine with [remote_ikernel](https://github.com/tdaff/remote_ikernel) to generate remote kernels from these specs.

Zsailer commented on docs/team/becoming-member.md:
```diff
@@ -0,0 +1,68 @@
+
+This means an inactive team member can "reactivate" themselves at any time by publicly stating their change in status. This does not require a nomination from another team member.
+
+For example, a team member who is going out on a long leave/vacation can temporarily move to the inactive team during their absence and immediately reactivate upon return.
```
```suggestion
For example, a team member who is going out on a long leave/vacation (>2 weeks) can temporarily move to the inactive team during their absence and immediately reactivate upon return. This isn't required, but this can relieve them from having to watch this repository for any formal votes that happen during their absence. 
```

Zsailer commented on docs/team/member-guide.md:
```diff
@@ -0,0 +1,111 @@
+4. **Let us know if you'll be unavailable** or out of town for an extended period
+   of time. It's no problem if you need to focus on other things for a bit, but it's
+   helpful for the team to know who will be around.
+   To do so, [open an issue in the team-compass](https://github.com/jupyter-server/team-compass/issues/new).
```
```suggestion
```
